test/c/test_sabre_layout.c

```diff
@@ -0,0 +1,325 @@
+            qk_circuit_instruction_clear(res_inst);
+            qk_circuit_instruction_clear(expected_inst);
+            free(res_inst);
+            free(expected_inst);
+            return false;
+        }
+        for (uint32_t j = 0; j < res_inst->num_qubits; j++) {
+            if (res_inst->qubits[j] != expected_inst->qubits[j]) {
+                printf("Qubit %d for gate %d are different %d was found and expected %d\n", j, i,
+                       res_inst->qubits[j] != expected_inst->qubits[j]);
+                printf("Expected circuit instructions:\n");
+                print_circuit(expected);
+                printf("Result circuit:\n");
+                print_circuit(res);
+                qk_circuit_instruction_clear(res_inst);
+                qk_circuit_instruction_clear(expected_inst);
+                free(res_inst);
+                free(expected_inst);
+                return false;
+            }
+        }
+        if (res_inst->num_clbits != expected_inst->num_clbits) {
+            printf("Gate %d have different number of clbits %d was found and expected %d", i,
+                   res_inst->num_clbits, expected_inst->num_clbits);
+            qk_circuit_instruction_clear(res_inst);
+            qk_circuit_instruction_clear(expected_inst);
+            free(res_inst);
+            free(expected_inst);
+            return false;
+        }
+        for (uint32_t j = 0; j < res_inst->num_clbits; j++) {
+            if (res_inst->clbits[j] != expected_inst->clbits[j]) {
+                printf("Clbit %d for gate %d are different %d was found and expected %d", j, i,
+                       res_inst->clbits[j] != expected_inst->clbits[j]);
+                qk_circuit_instruction_clear(res_inst);
+                qk_circuit_instruction_clear(expected_inst);
+                free(res_inst);
+                free(expected_inst);
+                return false;
+            }
+        }
+        if (res_inst->num_params != expected_inst->num_params) {
+            printf("Gate %d have different number of params %d was found and expected %d", i,
+                   res_inst->num_params, expected_inst->num_params);
+            qk_circuit_instruction_clear(res_inst);
+            qk_circuit_instruction_clear(expected_inst);
+            free(res_inst);
+            free(expected_inst);
+            return false;
+        }
+        for (uint32_t j = 0; j < res_inst->num_params; j++) {
+            if (res_inst->params[j] != expected_inst->params[j]) {
+                printf("Parameter %d for gate %d are different %d was found and expected %d", j, i,
+                       res_inst->params[j] != expected_inst->params[j]);
+                qk_circuit_instruction_clear(res_inst);
+                qk_circuit_instruction_clear(expected_inst);
+                free(res_inst);
+                free(expected_inst);
+                return false;
+            }
+        }
```

Comment:
It's not going to matter in this test suite, but this is asking for floating-point errors.